src/name_service/delivery/delete.clj

```diff
@@ -0,0 +1,34 @@
+(ns name-service.delivery.delete
+  (:require [cheshire.core :as json]
+            [clojure.spec.alpha :as s]
+            [name-service.delivery.http :as http]
+            [name-service.delivery.spec :as spec]
+            [name-service.delivery.state :as state]
+            [name-service.delivery.utils :as utils]))
+
+
+(defn- delete-keymap [kv]
+  (let [key (utils/map-key-first kv)
+        value (utils/map-value-first kv)]
+    (utils/action->response (state/delete-keymap key value))))
+
+(defn- delete-by-key [key-delete kv]
+  (let [key (utils/map-key-first kv)
+        value (utils/map-value-first kv)]
+    (-> key-delete
+        (state/delete-key-binding key value)
+        (utils/action->response))))
+
+(defn- delete-key-binding [json-params]
+  (if-let [key (:delete json-params)]
```

Comment:
I would add comment here, it's not trivially obvious what you're doing here in this `if-let`.